zone/zonedb.cpp

```diff
@@ -994,6 +1040,55 @@ bool ZoneDatabase::SaveCharacterLootLockout(uint32 character_id, uint32 expiry,
 	return true;
 }
 
+bool ZoneDatabase::SaveCharacterInstanceLockout(uint32 character_id, uint32 expiry, uint32 zone_id, uint32 zone_instance_id)
+{
+	std::string query = StringFormat("REPLACE INTO `character_instance_lockouts` (character_id, expiry, zone_id, zone_instance_id) VALUES (%u, %u, %u, %u)", character_id, expiry, zone_id, zone_instance_id);
+	auto results = QueryDatabase(query);
```

Comment:
What if this fails to save, should it handle that?

Reply:
We don't care if this fails to save, really - if it fails to save, we're in a catastrophic, world-ending scenario where the DB isn't present. Lockouts aren't a concern then.